File: lib/models/seed_data.dart
import 'package:google_maps_flutter/google_maps_flutter.dart';

class HangoutSeed {
  final String id;
  final String title;
  final LatLng position;
  final String category;
  final String time;

  HangoutSeed({
    required this.id,
    required this.title,
    required this.position,
    required this.category,
    required this.time,
  });
}

final List<HangoutSeed> SEEDED_HANGOUTS = [
  HangoutSeed(
    id: '1',
    title: 'Chai + Walk at Cubbon Park',
    position: const LatLng(12.9716, 77.5946),
    category: 'Walks',
    time: '4:00 PM',
  ),
  HangoutSeed(
    id: '2',
    title: 'Board Games at Third Wave',
    position: const LatLng(12.9780, 77.6408),
    category: 'Games',
    time: '7:00 PM',
  ),
  HangoutSeed(
    id: '3',
    title: 'Quick Cricket at HSR',
    position: const LatLng(12.9141, 77.6411),
    category: 'Sports',
    time: '6:30 PM',
  ),
];

File: utils/seedData.ts
import { geohashForLocation } from 'geofire-common';

export interface Hangout {
  id: string;
  title: string;
  description: string;
  category: string;
  location: {
    lat: number;
    lng: number;
    geohash: string;
  };
  startTime: string;
  endTime: string;
  hostId: string;
  participants: string[];
  capacity: number;
  meetingPoint?: string;
}

export const SEEDED_HANGOUTS: Hangout[] = [
  {
    id: '1',
    title: 'Chai + Walk at Cubbon Park',
    description: 'A casual stroll with some refreshing tea.',
    category: 'Social',
    location: {
      lat: 12.9757,
      lng: 77.5929,
      geohash: geohashForLocation([12.9757, 77.5929]),
    },
    startTime: new Date(Date.now() + 2 * 60 * 60 * 1000).toISOString(),
    endTime: new Date(Date.now() + 4 * 60 * 60 * 1000).toISOString(),
    hostId: 'host_1',
    participants: ['user_1', 'user_2'],
    capacity: 10,
    meetingPoint: 'Gate 2, Cubbon Park',
  },
  {
    id: '2',
    title: 'Board Games at Third Wave',
    description: 'Playing Catan and drinking coffee.',
    category: 'Gaming',
    location: {
      lat: 12.9345,
      lng: 77.6101,
      geohash: geohashForLocation([12.9345, 77.6101]),
    },
    startTime: new Date(Date.now() + 5 * 60 * 60 * 1000).toISOString(),
    endTime: new Date(Date.now() + 8 * 60 * 60 * 1000).toISOString(),
    hostId: 'host_2',
    participants: [],
    capacity: 6,
    meetingPoint: 'Corner table near the window',
  },
  {
    id: '3',
    title: 'Street Photography Loop',
    description: 'Exploring the vibrant streets of Indiranagar.',
    category: 'Creative',
    location: {
      lat: 12.9719,
      lng: 77.6412,
      geohash: geohashForLocation([12.9719, 77.6412]),
    },
    startTime: new Date(Date.now() + 24 * 60 * 60 * 1000).toISOString(),
    endTime: new Date(Date.now() + 27 * 60 * 60 * 1000).toISOString(),
    hostId: 'host_3',
    participants: ['user_4'],
    capacity: 5,
    meetingPoint: 'Indiranagar Metro Station',
  },
];
